python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -252,7 +264,19 @@ def _task_metadata(self):
         cluster = response.get("Cluster")
         task_arn = response.get("TaskARN")
 
-        task = self.ecs.describe_tasks(tasks=[task_arn], cluster=cluster)["tasks"][0]
+        def describe_task_or_raise(task_arn, cluster):
+            return self.ecs.describe_tasks(tasks=[task_arn], cluster=cluster)["tasks"][0]
+
+        try:
+            task = backoff(
+                describe_task_or_raise,
+                retry_on=(IndexError,),
```

Comment:
I think its worth leaving a comment here - this retry on IndexError to capture tasks/list not being populated yet pattern takes a second do discern without context

Reply:
Better yet, I'll explicitly raise that no tasks are found: https://github.com/dagster-io/dagster/pull/4448/commits/f34253c1123084faee4857e5b5c3f1195e0da5b5